magefile.go

```diff
@@ -1,17 +1,510 @@
+	}
+	must.Command("go", args...).CollapseArgs().RunV()
+
+	// Verify integration tests compile
+	must.Run("go", "test", "-run=non", "./tests/...")
+}
+
+func Build() {
+	rebuild, err := target.Dir(binDir, srcDirs...)
+	if err != nil {
+		mgx.Must(fmt.Errorf("error inspecting source dirs %s: %w", srcDirs, err))
+	}
+	if rebuild {
+		mg.SerialDeps(Clean, TestUnit)
+		//TODO: fix issue with XBuildAll https://github.com/getporter/magefiles/issues/4
+		var g errgroup.Group
+		for _, goos := range supportedClientGOOS {
+			goos := goos
+			g.Go(func() error {
+				return XBuild(pluginPkg, pluginName, binDir, goos, "amd64")
+			})
+		}
+		mgx.Must(g.Wait())
+		info := LoadMetadata()
+		os.RemoveAll(filepath.Join(binDir, "dev"))
+		shx.Copy(filepath.Join(binDir, info.Version), filepath.Join(binDir, "dev"), shx.CopyRecursive)
+		//copy local arch bin to bin/plugins/kubernetes/kubernetes for local integration testing
+		shx.Copy(filepath.Join(binDir, info.Version, fmt.Sprintf("%s-%s-%s", pluginName, runtime.GOOS, "amd64")), filepath.Join(binDir, pluginName))
+		PreparePluginForPublish(pluginName)
+	}
+}
+
+func SetupLocalTestEnv() {
+	SetupTests()
+}
+
+// Run local integration tests against the test cluster.
+func TestLocalIntegration() {
+	mg.SerialDeps(Build, SetupLocalTestEnv)
+	ctx, _ := kubectl("config", "current-context").OutputV()
+	testLocalIntegration()
+	must.RunV("go", "test", "-v", "./tests/integration/local/...")
+	if ctx != "" {
+		kubectl("config", "use-context", ctx).RunV()
+	}
+}
+
+func testLocalIntegration() {
+	kubectl("config", "use-context", localTestKubernetesContext).RunV()
+	kubectl("create", "namespace", localTestNamespace).RunV()
+	defer func() {
+		kubectl("delete", "namespace", localTestNamespace).RunV()
+	}()
+	kubectl("create", "secret", "generic", "password", "--from-literal=credential=test", "--namespace", localTestNamespace).RunV()
+	shx.Copy(filepath.Join(pwd(), "tests/integration/local/scripts/config-secret-test-local.toml"), filepath.Join(porterHome, "config.toml"))
+	kubectl("apply", "-f", filepath.Join(pwd(), "tests/testdata/credentials-secret.yaml"), "-n", localTestNamespace).RunV()
+	porter("plugins", "list").RunV()
+	porter("credentials", "apply", filepath.Join(pwd(), "tests/testdata/kubernetes-plugin-test-secret.json")).RunV()
+	porter("install", "--force", "--cred", "kubernetes-plugin-test", "-f", filepath.Join(pwd(), "tests/testdata/porter.yaml"), "--debug", "--debug-plugins").RunV()
+}
+
+// Run integration tests against the test cluster.
+func TestIntegration() {
+	mg.Deps(Build, EnsureGinkgo)
+	mg.SerialDeps(SetupTests, EnsureTestNamespace)
+	if os.Getenv("PORTER_AGENT_REPOSITORY") != "" && os.Getenv("PORTER_AGENT_VERSION") != "" {
+		localAgentImgRepository = os.Getenv("PORTER_AGENT_REPOSITORY")
+		localAgentImgVersion = os.Getenv("PORTER_AGENT_VERSION")
+	}
+	shx.Command("ginkgo").Args("-p", "-nodes", "4", "-v", "./tests/integration/operator/ginkgo", "-coverprofile=coverage-integration.out").
```

Comment:
The error from this command isn't being handled. If you want it to fail immediately just replace shx.Command with must.Command. Otherwise you'll want to check the error and handle it.